runtime/src/iree/hal/local/elf/elf_module.c

```diff
@@ -579,6 +580,12 @@ iree_status_t iree_elf_module_initialize_from_memory(
   IREE_ASSERT_ARGUMENT(out_module);
   IREE_TRACE_ZONE_BEGIN(z0);
 
+  // If the file is a FatELF then select the ELF for this architecture.
+  // Ignored of not a FatELF and otherwise errors if no compatible architecture
+  // is available.
+  IREE_RETURN_AND_END_ZONE_IF_ERROR(z0,
+                                    iree_fatelf_select(raw_data, &raw_data));
```

Comment:
What sorts of tests would be useful for this new code? Fuzz tests?

Reply:
I'd love to have someone set up ossfuzz - that'd be useful for this (and the elf loader in general) - it's kind of mostly for correctness/coverage, though, as once you're loading executable pages the thing loaded can do literally whatever it wants so security is less important (unless we add codesigning support, which would be cool).